conv/convert_format_test.go

```diff
@@ -0,0 +1,393 @@
+			str := FormatInteger(f)
+			c1, err := ConvertInt8(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertInteger[int8](str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidInts {
+			_, err := ConvertInt8(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertInteger[int8](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+
+	t.Run("with int16", func(t *testing.T) {
+		validInts := []int16{0, 1, -1, math.MaxInt8, math.MinInt8, math.MaxInt16, math.MinInt16}
+		invalidInts := []string{"1.233", "a", "false", strconv.FormatInt(int64(math.MaxInt64), 10)}
+
+		for _, f := range validInts {
+			str := FormatInteger(f)
+			c1, err := ConvertInt16(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertInteger[int16](str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidInts {
+			_, err := ConvertInt16(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertInteger[int16](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+
+	t.Run("with int32", func(t *testing.T) {
+		validInts := []int32{0, 1, -1, math.MaxInt8, math.MinInt8, math.MaxInt16, math.MinInt16, math.MinInt32, math.MaxInt32}
+		invalidInts := []string{"1.233", "a", "false", strconv.FormatInt(int64(math.MaxInt64), 10)}
+
+		for _, f := range validInts {
+			str := FormatInteger(f)
+			c1, err := ConvertInt32(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertInteger[int32](str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidInts {
+			_, err := ConvertInt32(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertInteger[int32](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+
+	t.Run("with int64", func(t *testing.T) {
+		validInts := []int64{0, 1, -1, math.MaxInt8, math.MinInt8, math.MaxInt16, math.MinInt16, math.MinInt32, math.MaxInt32, math.MaxInt64, math.MinInt64}
+		invalidInts := []string{"1.233", "a", "false"}
+
+		for _, f := range validInts {
+			str := FormatInteger(f)
+			c1, err := ConvertInt64(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
+			c2, err := ConvertInt64(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, c1, c2)
+		}
+
+		for _, f := range invalidInts {
+			_, err := ConvertInt64(f)
+			require.Error(t, err, testErrMsg(f))
+
+			_, err = ConvertInteger[int64](f)
+			require.Error(t, err, testErrMsg(f))
+		}
+	})
+}
+
+func TestConvertUinteger(t *testing.T) {
+	t.Run("with uint8", func(t *testing.T) {
+		validInts := []uint8{0, 1, math.MaxUint8}
+		invalidInts := []string{"1.233", "a", "false", strconv.FormatUint(math.MaxUint64, 10), strconv.FormatInt(-1, 10)}
+
+		for _, f := range validInts {
+			str := FormatUinteger(f)
+			c1, err := ConvertUint8(str)
+			require.NoError(t, err)
+			assert.EqualValues(t, f, c1)
+
```

Comment:
It should be this, I think


```suggestion
			assert.Equal(t, f, c1)

```

EqualValues compares the values, but not the type, here you would like to compare the type and values, no?

Reply:
Latest version of testifylint available with golangci-lint v2 reports it

https://github.com/Antonboom/testifylint/pull/223

Reply:
done